### DISCUSSION TITLE
It isn't clear how to stop a "currently playing session"

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [music-assistant/support#2030] [Feature requests and ideas]
From Discord:
> I tried looking it up but could not find. is there any way to end/cancel a currently playing session that is paused?
> ![CleanShot_2024-03-19_at_11 48 092x](https://github.com/music-assistant/support/assets/30210785/697af1b4-f468-4d1a-a530-3166ad84c024)

The user discovered that you need to clear the queue for the player, however this isn't obvious from the UI.

Could have a "clear queue" or "stop playing" button or something similar on the "currently playing" list?

**Comment by OzGav** [comment]
That isn’t part of the current UI design. We have docs which show the queue menu and the clear queue item

**Comment by JeffersonBledsoe** [comment]
@OzGav That's why I filed the issue, in order to make this operation clearer. While yes, clicking on the currently playing item, opening the queue and clearing it to remove it from this last makes sense if you're aware of the link between queued items and the currently playing list, it's not immediately obvious to somebody who has just started using MA, has noticed that, while a player might not be actively playing music (like if they paused it 20 minutes ago), it still shows under currently playing and they want to fully stop it.

The current behaviour when you click on the body (not the play/ pause) button for one of these 'currently playing' items is to select it as your active player. What if we showed a modal that gave the option to "select as current player", but also a shortcut to "clear queue" for that player? The modal could open up further 'quick actions' too.

**Reply by OzGav** [reply]
Not a crazy idea. As you know we don’t have a dedicated front end dev but if you wanted to put a PR together we can see what the others think…?

**Reply by marcelveldt** [reply]
Maybe just a power button?
A player does not have to be stopped! That is all handled automatically

**Reply by JeffersonBledsoe** [reply]
A power button would work, but we don't want to make it too easy for somebody to accidentally stop their music stream (e.g. the power button is close to the play/ pause). Do you think a power + a confirmation modal would work better than the 'body' of the 'currently playing' block opening a modal?

**Reply by OzGav** [reply]
I am still trying to understand why it is necessary?

**Reply by JeffersonBledsoe** [reply]
It's more of a usability improvemenet than a neccessary function. How will people who want to stop a "current playing" player think how to stop it? Will they think to switch their current player to the one they want to stop and clear it's queue, or will they just want to see there is a stream playing and press 'stop' and have the system figure out what to do.

**Reply by OzGav** [reply]
It is possible to pause. Why is that no good? Not trying to be argumentative just trying to understand as it never occured to me that this is needed? By pausing then your queue is ready to go again next time. What I think might be missing is the replace option when you want to play something. You can do that with a service call but from the UI there isn't the option to clear the queue and play now.

**Reply by JeffersonBledsoe** [reply]
You're not being argumentative at all! If you pause the other player, it will still remain in the "Currently playing" won't it? For me, 'stopping' is a way of saying "I'm done playing on this speaker/ group now", rather than coming back to it later. And so I don't want to see it on the "Currently playing" anymore as I don't intend on resuming it in anyway.

I do also like the replace idea!

**Reply by marcelveldt** [reply]
- If a queue is stopped, it should not be listed in that "now playing list" --> that would be a bug then.

- If you pause a queue, it will automatically be stopped after 30 seconds.

- I don't think we should add more mess here. It is a very common pattern to just have a play and pause button, please stop thinking in cassette deck era please. Open the spotify, sonos or whatever app and look for a stop button there, you won't find it.

- If you want the default action to be replace when you play media type X, you can select that preference in the player queue settings.

**Reply by Rudd-O** [reply]
> please stop thinking in cassette deck era please.

:sob:

**Comment by OzGav** [comment]
A STOP menu item in the NOW PLAYING view is available in 2.4 

![image](https://github.com/user-attachments/assets/a0139b9e-d832-4a5a-9d08-7ae53435687c)

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
